src/techui_builder/screen.py

```diff
@@ -80,73 +75,87 @@ def default_if_none(value: str | None) -> int:
-                    for i in range(len(self.screen_components)):
-                        if order == STACK_GLOBAL * i:
-                            group_width = group_padding
+                    # Reset X position after STACK_GLOBAL widgets
+                    if widget_count % STACK_GLOBAL == 0:
+                        widget_x = 0
 
                     # Add macros to the widgets
                     widgets[order].macro(self.P, ui.P)
                     widgets[order].macro(self.M, ui.M)
 
             if screen[ui.type]["type"] == "related":
-                # TODO: Add grouping action for related screens
+                height, width = (40, 100)
                 widgets.append(
                     Widget.ActionButton(
                         name,
                         name,
                         "{self.P:self.M}",
-                        (70 * order),
-                        (50 * math.floor(order / STACK_GLOBAL)),
-                        60,
-                        40,
+                        widget_x,
+                        0,
+                        width,
+                        height,
                     )
                 )
+                widget_x += width
+                widget_count += 1
+
+                # Reset X position after STACK_GLOBAL widgets
+                if widget_count % STACK_GLOBAL == 0:
+                    widget_x = 0
 
                 # Add action to action button: to open related display
                 widgets[order].action_open_file(
                     f"./techui-support/bob/{screen[ui.type]['file']}"
                 )
 
-        # Add widgets to groups
-        start_widget = 0
-        end_widget = STACK_GLOBAL
-        for group in groups:
-            group.add_widget(widgets[start_widget:end_widget])
-            start_widget += STACK_GLOBAL
-            end_widget += STACK_GLOBAL
+        widget_list = [
+            widgets[i : i + STACK_GLOBAL] for i in range(0, len(widgets), STACK_GLOBAL)
+        ]
+
+        # Make groups
+        def getGroupHeightandWidth(list):
+            """
+            Takes in a list of widget screens and finds the
+            maximum height in the list
+            """
+            height = []
+            width = []
+            for widget in list:
+                root = ET.fromstring(str(widget))
+                height_1 = root.find("height")
+                if height_1 is not None:
+                    height.append(default_if_none(height_1.text))
+
+                width_1 = root.find("width")
+                if width_1 is not None:
+                    width.append(default_if_none(width_1.text))
+
+            return [max(height) + 40, sum(width) + 40]
+
+        group_heights_widths = []
+
+        for lists in widget_list:
+            group_heights_widths.append(getGroupHeightandWidth(lists))
+
+        print(group_heights_widths)
+        stack_height = 0
+        for no, list in enumerate(group_heights_widths):
```

Comment:
what is `list` a list of? Can this be renamed? And can `no` be something like `id` instead to be a bit more programmatic?